### ISSUE TITLE
Unparseable date from GitLab webhook

### ISSUE THREAD

**Issue by MuyBien** [jenkinsci/gitlab-plugin#634]
## Issue

Hi!
I'm trying to link GitLab CE and Jenkins with a webhook. I follow his tutorial: https://medium.com/@teeks99/continuous-integration-with-jenkins-and-gitlab-fa770c62e88a

But in the final part, when I test to fire a webhook in GitLab, I get a error message saying that the date is not in a supported format.

I'm wondering if it's still possible to link GitLab and Jenkins with webhooks. Do I have to go to the GitLab EE and use the Jenkins intégration instead of a webhook ?

### Context
- **Gitlab plugin version**: 1.4.8
- **Gitlab version**: 10.0.1
- **Jenkins version**: 2.60.3
- **Job type**: Freestyle

### Logs & Traces

```
java.io.IOException: Unparseable date: "2017-03-08T16:38:33.870+01:00". Supported formats: [yyyy-MM-dd HH:mm:ss Z, yyyy-MM-dd'T'HH:mm:ss.SSS'Z', yyyy-MM-dd'T'HH:mm:ssX] 
at com.dabsquared.gitlabjenkins.util.JsonUtil$DateModule$1.deserialize(JsonUtil.java:66) 
at com.dabsquared.gitlabjenkins.util.JsonUtil$DateModule$1.deserialize(JsonUtil.java:53) 
```

### Problem description

_Describe your problem in a meaningful way_: Configure a webhook link between GitLab CE and Jenkins
- what were you doing (simple push, merge request, MR with fork, ...): Final webhook test
- what was expected: A success message and the job initialized
- what occurred finally: Jenkins return an error message

**Comment by nathanmauro** [comment]
Having the same issue

**Comment by lexapi** [comment]
Jenkins ver. 2.73.1
GitLab 10.0.3
GitLab API v4
Gitlab plugin version: 1.4.9

I fixed the problem with the date, but the integration still does not work (there is no connection between merge request and job in jenkins)

**Comment by omehegan** [comment]
Looks like GitLab may have changed the date format for some users... @dblessing can you comment on that?

@lexapi were you going to submit a PR with your fix?

**Comment by lexapi** [comment]
I can do it, but it did not help for me

**Comment by omehegan** [comment]
@lexapi looks like you might need to change yyyy-MM-dd'T'HH:mm:ss.SSSZ to yyyy-MM-dd'T'HH:mm:ss.SSS'+'Z

**Comment by lexapi** [comment]
My commit looks like this 63341d4bcbb03a45141a53edd8e157e1225ce8e4 (#653) i.e. he is already merged
Now we use gitlab 10.3.3 (in Docker) with the following settings for env:

> TZ Europe / Moscow
> GITLAB_TIMEZONE in Москва

and Jenkins ver. 2.89.2

I just updated the plug-in to 1.5.2, and everything works fine, i.e. The issue with dates has already been fixed.

But I found another problem in the plugin. He does not understand the redirect from HTTP to HTTPS, if gitlab running on HTTPS but in credential settings HTTP URL. After pressing Test button in settings i got message Client error: HTTP 302 Found
It would be nice to fix redirects and add errors logging to job output, for faster debugging
